src/main/java/org/elasticsearch/cluster/metadata/MappingMetaData.java

```diff
@@ -354,7 +354,7 @@ private void initMappers(Map<String, Object> withoutType) {
                     path = fieldNode.toString();
                 } else if (fieldName.equals("format")) {
                     format = fieldNode.toString();
-                } else if (fieldName.equals("default")) {
+                } else if (fieldName.equals("default") && fieldNode != null) {
```

Comment:
logic subscribing to what is described in the docs (ignoring null assigned to any other properties) would be more like this...  making defaultTimestamp null if so, else the string provided.

```
else if (fieldName.equals("default")) {
      defaultTimestamp = (fieldNode == null ? null : fieldNode.toString());
}
```